### PR TITLE
Integrate audio and some fixes

### PR DESCRIPTION
**Pull request by siueric0010**
---Extra ---
1. Audio included with selecting
2. Character textures are loaded before the game starts to ensure quicker select screen. (but now init takes >100 seconds, we need to optimize loading textures).
3. Directional Audio (not 3D directional, only stereo) on effects (i.e. attacks).

---Original Goals---
Audio
- Integrate Audio System
- Play intro music at jobs + loading
- Play fight music when players take damage

Misc
- Fix Player y positions
- Fix window crashing problem
- Add dry grass texture as default then compile the tiles in the mockup.

List of current audio:
    TITLE_MUSIC,
    COMBAT_MUSIC,
    FIGHTER_ATTACK_SOUND,
    MAGE_FIREBALL_SOUND,
    MAGE_ATTACK_SOUND,
    MONSTER_THROW_SOUND,
    CLERIC_ATTACK_SOUND,
    ROGUE_ATTACK_SOUND,
    SELECT_SOUND,

### CODE REVIEW

File: client/CSE125Client.vcxproj [hunk 1]
```diff
@@ -103,13 +103,42 @@
     <Link>
       <AdditionalDependencies>opengl32.lib;kernel32.lib;user32.lib;gdi32.lib;winspool.lib;comdlg32.lib;advapi32.lib;shell32.lib;ole32.lib;oleaut32.lib;uuid.lib;odbc32.lib;odbccp32.lib;%(AdditionalDependencies)</AdditionalDependencies>
     </Link>
+    <ClCompile>
+      <PreprocessorDefinitions>WITH_MINIAUDIO</PreprocessorDefinitions>
```
Comment (darylnak) [inline client/CSE125Client.vcxproj:107]: Add this file to gitignore?

File: client/CSE125Client.vcxproj.filters [hunk 2]
```diff
@@ -13,6 +13,9 @@
       <UniqueIdentifier>{67DA6AB6-F800-4c08-8B7A-83BB121AAD01}</UniqueIdentifier>
       <Extensions>rc;ico;cur;bmp;dlg;rc2;rct;bin;rgs;gif;jpg;jpeg;jpe;resx;tiff;tif;png;wav</Extensions>
     </Filter>
+    <Filter Include="Header Files\soloud">
```
Comment (darylnak) [inline client/CSE125Client.vcxproj.filters:16]: Add this file to gitignore?

**Review by darylnak — CHANGES_REQUESTED** [review]

File: client/CSE125Client.vcxproj (hunk 1 above)
Comment (siueric0010) [reply client/CSE125Client.vcxproj:107]: required to build the audio program

File: client/CSE125Client.vcxproj.filters (hunk 2 above)
Comment (siueric0010) [reply client/CSE125Client.vcxproj.filters:16]: required to build the audio program

**Review by darylnak — APPROVED** [review]
Good to go

**Pull request merged**